.github/workflows/upload_test_results_to_biquery.yaml

```diff
@@ -77,12 +77,25 @@ jobs:
           pip install -r requirements.txt
           pip install --upgrade google-cloud-bigquery
       - name: Download Artifact
+        id: download-artifact
+        continue-on-error: true
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ inputs.artifact }}
+          # gotta put this where the script will be run from
+          path: ${{ env.CHECKOUT_PATH }}${{ env.SCRIPT_PATH }}${{ env.ARTIFACT_PATH }}
+      # try using old v3 artifacts if newest fails
+      - name: Download Artifact Legacy v3
+        id: download-artifact-legacy-v3
+        if: ${{ steps.download-artifact.outcome == 'failure' }}
+        continue-on-error: true
         uses: actions/download-artifact@v3
         with:
           name: ${{ inputs.artifact }}
           # gotta put this where the script will be run from
           path: ${{ env.CHECKOUT_PATH }}${{ env.SCRIPT_PATH }}${{ env.ARTIFACT_PATH }}
       - name: Auth to GCP
+        if: ${{ steps.download-artifact-legacy-v3.outcome == 'success' || steps.download-artifact.outcome == 'success' }}
```

Comment:
**question:** do we need to have this if statement on all subsequent steps?

Reply:
`continue-on-error: true` forces the `conclusion` to `success`, which is what determines the check or X. The `outcome` of the step is still `failure` though.

Techinically we _don't_ need this if at all if both previous `outcomes` were `failures`, it'd just auth correctly and then fail the upload b/c there's no file to upload.